api/client/options.proto

```diff
@@ -14,6 +14,19 @@ message RequestOptions {
   google.protobuf.UInt32Value request_body_size = 3 [(validate.rules).uint32 = {lte: 4194304}];
 }
 
+// We wrap all values so we can have a unified way of option handling with respect to
+// defaults, merging, etc. As there's no stock concept for enumerations, we manually
+// define custom wrappers for them. These used to be strings, which did provide the
+// wrapped type.
```

Comment:
This is getting a bit clunky; can you remind me why we need to have this for all primitive types?

Reply:
Without wrapping, we cannot make a distinction between ‘type-default’ and ‘not-set’. Which interferes with some things, like when we want to merge two proto’s and some would like to override something with the type default. Wrapping allows us to make that distinction. I must admit I have not tried if enum types behave exactly like this, I now realize I made an assumption that they would be treated similar to integers at the serialization level.

Reply:
Yeah, I think enums are effectively scalars. Hmm, Do we need full merge capabilities for client options here? I feel that there is some set that is CLI and fixed forever and others we want to dynamically update via header overrides etc.

Reply:
There is the test-server, which could read a base config from disk, but maybe that would also be nice for the grpc service (and even the CLI)? That would allow the end-user to override the hard-coded defaults, and subsequently tune via cli/grpc calls/headers. With uniform behavior across the tooling that we offer.

Reply:
Trying to figure out which bits might never really change. E.g. could address family here change dynamically?

Reply:
Perhaps; maybe far fetched, but a grpc server could be configured to auto hard-coded or on disk, and then overridden to v4 or v6 via calls to initiate benchmark runs.. Not sure if this is realistic, but I'm also not sure if we should prevent it?

Reply:
After discussing this on Slack, I'm leaning towards keeping this as-is. I can see scenarios for any of the enum-fields where being able to override is going to be useful, and the way the proto is defined right now avoids having to change the api.